nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/executor/WorkflowDispatcher.java

```diff
@@ -55,6 +55,9 @@ public void run() {
 
           if (!shutdownRequested) {
             executorRecovery.tick();
+            if (stateProcessorFactory.getPotentiallyStuckProcessors() == executor.getThreadCount()) {
+              logger.warn("All state processor threads are potentially stuck.");
```

Comment:
Once things get stuck, do we need the logging at every polling interval? 

Maybe use this https://github.com/NitorCreations/nflow/blob/master/nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/util/PeriodicLogger.java